src/lib/Microsoft.Health.Fhir.Ingest/Service/FhirHealthService.cs

```diff
@@ -0,0 +1,14 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public abstract class FhirHealthService
```

Comment:
Thoughts on the public method just returning Task and taking an ILogger for telemetry recording?  In this instance you would still have version of CheckHealth returning the FhirHealthCheckStatus but it would be a protected abstract member that the implementer needs to specify.  Then the public method of Task Name(ILogger, CancelationToken) would record the result in a standard way in the logger so the upstream implementer doesn't need to.  Only thing you need in the function definition is the instance of the Health Check & the ILogger which you can get with DI on the function signature.  This also makes the use of the abstract class more appropriate.  Right now the abstract class  doesn't have any implementation outside a contract and could just be an interface.

Reply:
That makes sense, but I'm not sure I would like to address this at this moment

Reply:
Okay, it can be part of the next step.